scripts/flights2.py

```diff
@@ -0,0 +1,727 @@
+
+    def scan_sources(self) -> SourceMap:
+        logger.info("Scanning dependencies ...")
+        for spec in self:
+            self.sources.add_dependency(spec)
+        msg = f"Finished scanning {len(self.sources)!r} date ranges."
+        logger.info(msg)
+        return self.sources
+
+    def run(self) -> None:
+        logger.info("Starting job ...")
+        self.download_sources()
+        for spec, frame in self.scan_sources().iter_tasks():
+            result = spec.transform(frame)
+            spec.write(result, self.output_dir)
+        logger.info("Finished job.")
+
+
+async def _request_async(session: niquests.AsyncSession, name: str, /) -> io.BytesIO:
+    name = f"{_without_suffixes(name)}{ZIP}"
+    msg = f"Requesting {name!r} ..."
+    logger.info(msg)
+    async with session:
+        response = await session.get(name)
+        if response.ok and (content := response.content):
+            buf = io.BytesIO()
+            buf.write(content)
+            msg = f"Successful {name!r}"
+            logger.info(msg)
+            return buf
+        msg = f"Failed for {name!r}"
+        raise NotImplementedError(msg)
+
+
+def _write_rezip(input_dir: Path, buf: io.BytesIO, /) -> Path:
+    """
+    Extract inner csv from a zip file, writing to a gzipped csv of the same name.
+
+    Notes
+    -----
+    - ``.read_bytes()`` is the only expensive op here
+    - End result (gzip, single file) can be scanned in parallel by ``polars``
+        - And slightly smaller than zipped directory
+    """
+    zip_csv = next(zipfile.Path(zipfile.ZipFile(buf)).glob("*.csv"))
+    stem = zip_csv.at.replace("(", "").replace(")", "")
+    gzipped: Path = (input_dir / stem).with_suffix(".csv.gz")
+    gzipped.touch()
+    msg = f"Writing {gzipped.as_posix()!r}"
+    logger.debug(msg)
+    with gzip.GzipFile(gzipped, mode="wb", mtime=0) as f:
+        f.write(zip_csv.read_bytes())
+    return gzipped
+
+
+async def _write_rezip_async(input_dir: Path, buf: io.BytesIO, /) -> Path:
+    """
+    Wraps ``_write_rezip`` to run in a separate thread.
+
+    - **Greatly** reduces the cost of the decompress > compress operations
+    - During testing, each write would block for ~10s
+    """
+    return await asyncio.to_thread(_write_rezip, input_dir, buf)
+
+
+def _file_stem_source[T: (str, pl.Expr)](year: T, month: T, /) -> pl.Expr:
+    """Returns an expression that composes the file stem for a single month."""
+    return pl.concat_str(pl.lit(REPORTING_PREFIX), year, pl.lit("_"), month)
+
+
+def _without_suffixes[T: (str, Path)](source: T, /) -> T:
+    """Ensure all suffixes (not just the last) are removed."""
+    if isinstance(source, str):
+        return source.removesuffix("".join(Path(source).suffixes))
+    return Path(str(source).removesuffix("".join(source.suffixes)))
+
+
+def _clean_source(ldf: pl.LazyFrame, /) -> pl.LazyFrame:
+    """Parsing data types, dropping invalid data, dropping unused columns, fixing a datetime issue."""
```

Comment:
### TODO
- [ ] Rewrite doc
- [ ] Possibly rename

Reply:
Resolved in (https://github.com/vega/vega-datasets/pull/645/commits/a002c39cd31e8766b77ce1bab3bb3e2c8195c36d)